gdx/src/com/badlogic/gdx/maps/tiled/iterators/OrthogonalTiledMapIterator.java

```diff
@@ -0,0 +1,75 @@
+package com.badlogic.gdx.maps.tiled.iterators;
+
+import com.badlogic.gdx.graphics.OrthographicCamera;
+import com.badlogic.gdx.maps.tiled.TiledMapTileLayer;
+import com.badlogic.gdx.math.GridPoint2;
+import com.badlogic.gdx.math.Rectangle;
+import java.util.Iterator;
+
+/**
+ * Iterates over tiles in visible area using the same logic as OrthogonalTiledMapRenderer from com.badlogic.gdx.maps.tiled.renderers.
+ * Using this together with OrthogonalTiledMapRenderer makes it easy to draw things on top of tiles that are visible, without drawing to non-visible portions of the tile map
+ * Created by BenMcLean on 5/5/2016.
+ */
+public class OrthogonalTiledMapIterator implements Iterator<GridPoint2> {
+    protected int x, x1, x2, y, y1, y2;
+    protected OrthographicCamera camera;
+    protected TiledMapTileLayer layer;
+    protected float unitScale;
+    protected Rectangle viewBounds = new Rectangle();
+
+    /** Assumes unit scale of 1.0f */
+    public OrthogonalTiledMapIterator(OrthographicCamera camera, TiledMapTileLayer layer) {
+        this(camera, layer, 1.0f);
+    }
+
+    public OrthogonalTiledMapIterator(OrthographicCamera camera, TiledMapTileLayer layer, float unitScale) {
+        this.camera = camera;
+        this.layer = layer;
+        this.unitScale = unitScale;
+        reset();
+    }
+
+    public OrthogonalTiledMapIterator reset() {
```

Comment:
Sometimes you want just the coordinates, (like when you're drawing things on top of the tiled map which aren't part of the cells, such as some health bars I drew which actually draw some distance above the cells) and if you've got the coordinates then you should be able to get the cells I think.